### PR TITLE
Update latest version references when retracting/restoring

### PR DESCRIPTION
**Pull request by szakarias**

### CODE REVIEW

File: app/lib/package/models.dart [hunk 1]
```diff
@@ -235,12 +236,46 @@ class Package extends db.ExpandoModel<String> {
...
+  }
+
   /// Updates latest stable, prerelease and preview versions and published
   /// timestamp with the new version.
   void updateVersion(
     PackageVersion pv, {
     required Version dartSdkVersion,
+    bool includeRetracted = false,
   }) {
+    if (!includeRetracted && pv.isRetracted) return;
```
Comment (isoos) [inline app/lib/package/models.dart:278]: I'm wondering if this is needed here, or maybe at line 252? (see above)

File: app/lib/package/models.dart [hunk 2]
```diff
@@ -235,12 +236,46 @@ class Package extends db.ExpandoModel<String> {
...
+    final oldStableVersion = latestSemanticVersion;
+    final oldPrereleaseVersion = latestPrereleaseSemanticVersion;
+    final oldPreviewVersion = latestPreviewSemanticVersion;
+
+    // reset field values
+    latestVersionKey = null;
+    latestPrereleaseVersionKey = null;
+    latestPreviewVersionKey = null;
+
+    versions.forEach((v) => updateVersion(v,
```
Comment (isoos) [inline app/lib/package/models.dart:252]: `versions.where((pv) => !pv.isRetracted).forEach()` ?

File: app/test/package/retracted_test.dart [hunk 3]
```diff
@@ -164,5 +165,120 @@ void main() {
...
+      final origLatestPrereleasePublished = pkg.latestPrereleasePublished;
+
+      final client = createPubApiClient(authToken: adminAtPubDevAuthToken);
+
+      final orig = await client.getVersionOptions('oxygen', '1.2.0');
+      expect(orig.isRetracted, isFalse);
+      final origInfo = await client.packageVersionInfo('oxygen', '1.2.0');
+      expect(origInfo.isRetracted, isNull);
+
+// Retract latest
```
Comment (isoos) [inline app/test/package/retracted_test.dart:196]: Nit: these comments are oddly indented.

**Review by isoos — APPROVED** [review]
Looks good, a few nits to change.

File: app/lib/package/models.dart (hunk 1 above)
Comment (szakarias) [reply app/lib/package/models.dart:278]: Sure I can move that logic one level up, it's much nicer.

File: app/test/package/retracted_test.dart (hunk 3 above)
Comment (szakarias) [reply app/test/package/retracted_test.dart:196]: Done

**Pull request merged**